packages/dataviews/src/style.scss

```diff
@@ -223,27 +223,69 @@
 
 .dataviews-list-view {
 	margin: 0;
+	padding: $grid-unit-10;
 
 	li {
-		border-bottom: $border-width solid $gray-100;
 		margin: 0;
-		&:first-child {
-			border-top: $border-width solid $gray-100;
+
+		.dataviews-list-view__item-wrapper {
+			position: relative;
+			padding-right: $grid-unit-30;
+			border-radius: $grid-unit-05;
+
+			&::after {
+				position: absolute;
+				content: "";
+				top: 100%;
+				left: $grid-unit-30;
+				right: $grid-unit-30;
+				background: $gray-100;
+				height: 1px;
+			}
 		}
-		&:last-child {
-			border-bottom: 0;
+
+		&:not(.is-selected):hover {
+			color: var(--wp-admin-theme-color);
+
+			.dataviews-list-view__fields {
+				color: var(--wp-admin-theme-color);
+			}
 		}
 	}
 
-	.dataviews-list-view__item {
-		padding: $grid-unit-15 $grid-unit-40;
-		cursor: default;
-		&:focus,
-		&:hover {
-			background-color: lighten($gray-100, 3%);
+	li.is-selected,
+	li.is-selected:focus-within {
+		.dataviews-list-view__item-wrapper {
+			background-color: var(--wp-admin-theme-color);
+			color: $white;
+
+			.dataviews-list-view__fields,
+			.components-button {
+				color: $white;
+			}
+
+			&::after {
+				background: transparent;
+			}
 		}
+	}
+
+	.dataviews-list-view__item {
+		padding: $grid-unit-15 0 $grid-unit-15 $grid-unit-30;
+		width: 100%;
+		cursor: pointer;
 		&:focus {
-			box-shadow: inset 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color);
+			&::before {
```

Comment:
This pseudo-element serves to visualize the focus ring as it was part of the parent node (`.dataviews-list-view__wrapper`). See [conversation](https://github.com/WordPress/gutenberg/pull/57578/#discussion_r1442701932).